### PR TITLE
(maint) Convert named parameters to hash

### PR DESCRIPTION
**Pull request by smcclellan**
In order to be functional on JRuby 1.7.x, we cannot rely on features that are
only present in later releases of Ruby, like named parameters. This reverts some
code that used that new feature.

### CODE REVIEW

**Comment by smcclellan** [comment]
They are generated and deleted in every test, and somehow got committed to git. They could also be gitignore'd, in case a test stops halfway through. I'll add that now.

**Comment by eputnam** [comment]
They were probably there from before I added the generate/delete stuff. sorry i missed them!

File: lib/gettext-setup/pot.rb [hunk 1]
```diff
@@ -50,9 +50,14 @@ def self.string_changes?(old_pot, new_pot)
       return true
     end
 
-    def self.generate_new_pot(locales_path: GettextSetup.locales_path, target_path: nil)
+    # @param [:locales_path] opts
+    #   The directory for the locales.
+    # @param [:target_path] opts
+    #   The output path for the new POT file.
+    def self.generate_new_pot(opts = {})
```
Comment (eputnam) [inline lib/gettext-setup/pot.rb]: will these functions poop if no hash is passed in?

Comment (Magisus) [reply lib/gettext-setup/pot.rb]: No, because the `= {}` passes an empty hash by default. So `[]` will just return nil, which we handle.

Comment (eputnam) [reply lib/gettext-setup/pot.rb]: image

Comment (smcclellan) [reply lib/gettext-setup/pot.rb]: They should pass, filling in defaults below. Unless `nil` is passed in explicitly.

**Pull request merged**